Data/StaffRepository.cs

```diff
@@ -0,0 +1,77 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Data;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using stunning_robot_HR.Models;
+
+namespace stunning_robot_HR.Data
+{
+    public class StaffRepository : IStaffRepository, IDisposable
+    {
+        private stunning_robot_HRContext _context;
+
+        public StaffRepository(stunning_robot_HRContext context)
```

Comment:
Another little anti-pattern. Because most of your controllers interact with the database. If you see a constructor that isn't taking an interface for a repo, you're probably directly interacting with the context or creating a new repo in your controller. The point? It's a little sign that you're missing a repository to ensure your code stays loosely coupled.